### PR TITLE
add-controller-and-mapstruct

### PR DESCRIPTION
**Pull request by ihorromaniuk**
branch name with mistake

### CODE REVIEW

File: src/main/java/core/basesyntax/bookstore/BookstoreApplication.java [hunk 1]
```diff
@@ -20,29 +20,28 @@ public static void main(String[] args) {
...
-                System.out.println(bookService.save(bot));
-                System.out.println(bookService.findAll());
-            }
+        return args -> {
+            CreateBookRequestDto shining = new CreateBookRequestDto(
+                    "The Shining",
+                    "Stephen King",
+                    "9781444720723",
+                    BigDecimal.valueOf(645),
+                    "spooky book",
```
Comment (liliia-ponomarenko) [inline src/main/java/core/basesyntax/bookstore/BookstoreApplication.java:29]: you can remove it for now and use endpoints

File: src/main/java/core/basesyntax/bookstore/controller/BookController.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/books")
+@RequiredArgsConstructor
+public class BookController {
+    private final BookService bookService;
+
+    @GetMapping
+    public ResponseEntity<?> getAllBooks() {
```
Comment (liliia-ponomarenko) [inline src/main/java/core/basesyntax/bookstore/controller/BookController.java:25]: you can return an object without ResponseEntity wrapping

File: src/main/java/core/basesyntax/bookstore/controller/BookController.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        return ResponseEntity.ok(bookService.findAll());
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<?> getBookById(@PathVariable Long id) {
+        return ResponseEntity.ok(bookService.getById(id));
+    }
+
+    @PostMapping
```
Comment (liliia-ponomarenko) [inline src/main/java/core/basesyntax/bookstore/controller/BookController.java:34]: ResponseStatus.CREATED

File: src/main/java/core/basesyntax/bookstore/dto/book/CreateBookRequestDto.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package core.basesyntax.bookstore.dto.book;
+
+import core.basesyntax.bookstore.model.Book;
+import java.math.BigDecimal;
+
+/**
+ * DTO for {@link Book}
+ */
```
Comment (liliia-ponomarenko) [inline src/main/java/core/basesyntax/bookstore/dto/book/CreateBookRequestDto.java]: That is not an informative comment. Let’s  remove it

File: src/main/java/core/basesyntax/bookstore/mapper/BookMapper.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+import core.basesyntax.bookstore.config.MapperConfig;
+import core.basesyntax.bookstore.dto.book.BookDto;
+import core.basesyntax.bookstore.dto.book.CreateBookRequestDto;
+import core.basesyntax.bookstore.model.Book;
+import org.mapstruct.Mapper;
+
+@Mapper(config = MapperConfig.class)
+public interface BookMapper {
+    Book createRequestDtoToEntity(CreateBookRequestDto dto);
```
Comment (liliia-ponomarenko) [inline src/main/java/core/basesyntax/bookstore/mapper/BookMapper.java]:
```suggestion
    Book toModel(CreateBookRequestDto dto);
```

File: src/main/java/core/basesyntax/bookstore/mapper/BookMapper.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+    BookDto entityToDto(Book book);
```
Comment (liliia-ponomarenko) [inline src/main/java/core/basesyntax/bookstore/mapper/BookMapper.java]:
```suggestion
    BookDto toDto(Book book);
```

File: src/main/java/core/basesyntax/bookstore/service/impl/BookServiceImpl.java [hunk 7]
```diff
@@ -11,15 +14,26 @@
...
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::entityToDto)
+                .toList();
+    }
+
+    @Override
+    public BookDto getById(Long id) {
+        return bookMapper.entityToDto(bookRepository.findById(id).orElseThrow(() ->
+                new EntityNotFoundException("Can't find book by id. Id: " + id)));
```
Comment (liliia-ponomarenko) [inline src/main/java/core/basesyntax/bookstore/service/impl/BookServiceImpl.java:36]: better create your custom exception

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s  fix a few comments ;)

File: src/main/java/core/basesyntax/bookstore/BookstoreApplication.java (hunk 1 above)
Comment (ihorromaniuk) [reply src/main/java/core/basesyntax/bookstore/BookstoreApplication.java:29]: I'd like to keep it to have non empty DB on start-up, also going to do same thing with other entities for development period (I still do Postman testing when implementing something or changing entity related)

File: src/main/java/core/basesyntax/bookstore/controller/BookController.java (hunk 2 above)
Comment (ihorromaniuk) [reply src/main/java/core/basesyntax/bookstore/controller/BookController.java:25]: I'd like to keep it for aesthetics, if it's not crucial

**Review by okuzan — APPROVED** [review]

**Pull request merged**